qemu/tests/iperf_test.py

```diff
@@ -0,0 +1,173 @@
+                session.cmd(compile_cmd)
+            else:
+                logging.info("Compiling %s in host..." % iperf_version)
+                process.run(compile_cmd, shell=True, verbose=False)
+        except (process.CmdError, ShellCmdError) as err_msg:
+            logging.error(err_msg)
+            test.error("Failed to compile iperf")
+        else:
+            iperf_bin_name = re.sub(r'[-2]', '', iperf_version.split('.')[0])
+            return os.path.join(iperf_source_path, 'src', iperf_bin_name)
+
+    def iperf_start(session, iperf_path, options, catch_data):
+        """Start iperf session, analyze result if catch_data."""
+        iperf_cmd = iperf_path + options
+        try:
+            info_text = "Start iperf session in %s with cmd: %s"
+            if session:
+                logging.info(info_text % ("guest", iperf_cmd))
+                data_info = session.cmd_output(iperf_cmd, timeout=120)
+            else:
+                logging.info(info_text % ("host", iperf_cmd))
+                data_info = process.system_output(iperf_cmd, timeout=120,
+                                                  verbose=False).decode()
+        except Exception as err_msg:
+            logging.error(str(err_msg))
+            test.error("Failed to start iperf session")
+        else:
+            if catch_data:
+                logging.debug("Full connection log:\n%s" % data_info)
+                parallel_cur = len(re.findall(catch_data, data_info))
+                parallel_exp = int(params.get("parallel_num", 0))
+                if not parallel_cur:
+                    test.fail("iperf client not connected to server")
+                elif parallel_exp and parallel_cur != parallel_exp:
+                    test.fail("Number of parallel threads running(%d) is "
+                              "inconsistent with expectations(%d)"
+                              % (parallel_cur, parallel_exp))
+                logging.info("iperf client successfully connected to server")
+
+    def is_iperf_running(name_pattern, session=None):
+        if session:
+            check_iperf_cmd = params["check_iperf_cmd"] % name_pattern
+            status = serial_session.cmd_status(check_iperf_cmd)
+        else:
+            status = process.system("pgrep -f %s" % name_pattern,
+                                    ignore_status=True, verbose=False)
+        return status == 0
+
+    os_type = params["os_type"]
+    login_timeout = int(params.get("login_timeout", 360))
+    fw_stop_cmd = params["fw_stop_cmd"]
+    tmp_dir = params.get("tmp_dir", "/tmp/")
+    iperf_test_duration = int(params["iperf_test_duration"])
+    iperf_deps_dir = data_dir.get_deps_dir("iperf")
+    host_iperf_file = params["host_iperf_file"]
+    guest_iperf_file = params.get('guest_iperf_file', host_iperf_file)
+    host_src_path = os.path.join(iperf_deps_dir, host_iperf_file)
+    guest_src_path = os.path.join(iperf_deps_dir, guest_iperf_file)
```

Comment:
deps_dir is for host only.  It is strange to use it for guest.

Reply:
deps_dir (dependent dir), which saves some test tools for test cases. If we want to use local test tool, it is not strange to use it for guest.

Reply:
I see it is only a path,  any linux system can have it.  
However,  no need to create a same path in guest,  and it make code looks mess.

Reply:
For windows guest, iperf file is not the same, so `guest_src_path` is for all guests, and we don't have to distinguish what it is.

Reply:
> For windows guest, iperf file is not the same, so `guest_src_path` is for all guests, and we don't have to distinguish what it is.

I do not why you mention `iperf file is not the same for windows guest, so guest_src_path is for all guests`.  It is obvious file name and file path are not same for linux and window guest.
My point here is, for linux guest, no need to use this deps dir (although any path is ok).

When talking `deps`, we usually means host path, it is an existing path when setup autotest env. For guest, any others such as "/home", "/tmp" are all ok.

Reply:
What do you mean by not using the `deps_dir` in the guest?
the currect logs are:
```
2019-07-29 06:43:01,896 remote           L0521 DEBUG| Trying to SCP with command 'scp -r -v -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o PreferredAuthentications=password  -P 22 /home/yihyu/kar/workspace/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/deps/iperf/iperf-2.0.12.tar.gz root@\[10.16.213.234\]:/tmp/', timeout 600s
...
...
2019-07-29 06:43:12,765 client           L1132 DEBUG| Sending command: tar zxf /tmp/iperf-2.0.12.tar.gz -C /tmp/ > /dev/null ; cd /tmp/iperf-2.0.12 ; ./configure > /dev/null; make > /dev/null
```

Reply:
`/home/yihyu/kar/workspace/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/deps/iperf/iperf-2.0.12.tar.gz`   this is the path in host, not guest, but you use `guest_src_path`, although the value for these two params are same for linux guest.

Reply:
It means a path of guest iperf source file on host dir. However, if you still it isn't a good name, I already updated it.